includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -853,21 +853,41 @@ public function process_payment( $order_id ) {
 			}
 
 			if ( ! empty( $payment_information ) ) {
-				$note = sprintf(
-					WC_Payments_Utils::esc_interpolated_html(
-						/* translators: %1: the failed payment amount, %2: error message  */
+				if ( $e instanceof API_Exception && 'card_error' === $e->get_error_type() && 'incorrect_zip' === $e->get_error_code() ) {
```

Comment:
I'd suggest reducing to simplify, so we have original `sprintf` only, but generate the detailed message in if-else block. That would better express the change we are making here and make it easier to extend in the future.

Reply:
Makes complete sense. I have simplified the logic, thanks.